### ISSUE TITLE
Cyclical references in Input Types

### ISSUE THREAD

**Issue by clintburgos** [Youshido/GraphQL#123]
Hi everyone,

I want to have cyclical references in my input types, something like:
```
parent {
    child {
        parent {
            child {
                ...
            }
        }
    }
}
```

My types reference each other in their fields without complaint and work fine. However, when I try to create a mutation with input type equivalents referencing other input types in their fields, I reach maximum recursion depth. If I reference types within my input types fields I get an error about missing input types, but it all works fine-- only auto complete is broken.

I'm not sure if this is just something I'm doing wrong, but it could be an issue. I did some searching and it seems like lazily loading field definitions is the answer, which I couldn't find anything on for this library.

Anyways, let me know either way, and thanks for creating this library, it's been a great help :)

**Comment by Fgruntjes** [comment]
Having the exact same issue, have not find a solution yet.

**Comment by Fgruntjes** [comment]
This does not address the underlying issue but I ended up doing this wich seems to work

```php
class FilterInputType extends AbstractInputObjectType
{
    /**
     * @param ObjectTypeConfig $config
     */
    public function build($config)
    {
        $config->addFields([
            'children' => new ListType($this),
        ]);
    }
}
```

**Comment by clintburgos** [comment]
Any update on this? I have some 1:M relationships like Fgruntjes' example, but some are 1:1 and still breaking. Hoping for a lazy reference trick as described here for the JavaScript implementation: http://siawyoung.com/coding/javascript/circular-references-graphql-type-definitions.html

Thanks!

**Comment by viniychuk** [comment]
@clintburgos @Fgruntjes fixed in 1.4.3.2
